### PR TITLE
Features

### PR DESCRIPTION
**Pull request by Oluwadamilareolusakin**
# Description
Eventor is a private event management app

# Features
* User creation
* Authentication via emails(Users authenticate via emails)
* Event creation
* Past events for users
* Upcoming events for users
* Events index(For admins with past and upcoming event sections)
* Users index(For admins)

# How it works
* A user arrives at landing page 
* Logs in or signs up
* Is sent an email containing a link to login
* Clicks the link and gets into their dashboard 
#### After sign in they can:
* Manage their events
* View attendees for their events
* The can also create events
* View events they have attended
* View events they will attend
* Send out invitations(In progress)
* Manage their invitations(In progress)
* View events they've been invited to(In progress)

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class Event < ApplicationRecord
+  scope :upcoming, -> { where('event_date > ?', Date.today) }
+  scope :past, -> { where('event_date < ?', Date.today) }
+
+  validates :title, presence: true
+  validates :description, presence: true
+  validates :event_time, presence: true
+  validates :event_date, presence: true
```
[the remark this answers is not in the document]
Comment (Oluwadamilareolusakin) [reply app/models/event.rb]: Noted, with make the adjustment. Thank you

[the remark this answers is not in the document]
Comment (Oluwadamilareolusakin) [reply app/models/event.rb]: Done

**Comment by Oluwadamilareolusakin** [comment]
@Oghenebrume50 I've bricked the sign up action in the users controller and let you in.

**Comment by Oluwadamilareolusakin** [comment]
@Oghenebrume50 sign up and it'll log you in

File: spec/models/attendance_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Attendance, type: :model do
+  before :each do
+    @user = FactoryBot.create(:user)
```
Comment (adriaanbd) [inline spec/models/attendance_spec.rb]: Let FactoryBot know about your association so you won't need to create a `user` explicitly

File: spec/models/attendance_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    @event = FactoryBot.create(:event, planner: @user)
```
Comment (adriaanbd) [inline spec/models/attendance_spec.rb]: Let FactoryBot know about your association so you won't need to create an `event` explicitly

File: spec/models/event_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Event, type: :model do
+  before :each do
+    @planner = FactoryBot.create(:user)
```
Comment (adriaanbd) [inline spec/models/event_spec.rb]: Let FactoryBot know about your association so you won't need to create a `user` explicitly

File: spec/factories/events.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :event do
```
Comment (adriaanbd) [inline spec/factories/events.rb]: Add the associated model in here so FactoryBot creates an user automatically for every event created

File: spec/factories/users.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :user do
```
Comment (adriaanbd) [inline spec/factories/users.rb]: Let FactoryBot know about your user aliases (planner, attendee, etc)

File: app/views/layouts/_nav.html.erb [hunk 7]
```diff
@@ -0,0 +1,35 @@
+<% unless controller.action_name == 'new'  && controller.controller_name == 'users' || controller.action_name == 'create'|| controller.controller_name == 'sessions' || controller.action_name == 'confirmation'%>
```
Comment (adriaanbd) [inline app/views/layouts/_nav.html.erb]: A lot of logic in here. Consider moving this to a helper.
[`OPTIONAL`]

**Comment by Oluwadamilareolusakin** [comment]
@adriaanbd are you still available?

File: app/views/layouts/_nav.html.erb (hunk 7 above)
Comment (Oluwadamilareolusakin) [reply app/views/layouts/_nav.html.erb]: Noted with thanks

**Pull request merged**